addons/packages/fluxcd-source-controller/0.21.2/package.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  name: fluxcd.source.controller.community.tanzu.vmware.com.0.21.2
+spec:
+  refName: fluxcd.source.controller.community.tanzu.vmware.com
+  version: 0.21.2
+  releasedAt: "2022-02-07T11:14:08Z"
+  valuesSchema:
+    openAPIv3:
+      properties: {}
+  template:
+    spec:
+      fetch:
+      - imgpkgBundle:
+          image: projects.registry.vmware.com/tce/fluxcd-source-controller-bundle@sha256:020f9efbc07d439bc9bc1857abdca053a9470eb98ba5d5af8de1fa43db141c93
+      template:
+      - ytt:
+          inline:
+            paths:
+              secretgen.yaml: |
+                #@ load("@ytt:overlay", "overlay")
+                ---
+                apiVersion: v1
+                kind: Secret
+                metadata:
+                  name: canonical-registry-secret
+                  namespace: flux-system
+                  annotations:
+                    secretgen.carvel.dev/image-pull-secret: ""
+                type: kubernetes.io/dockerconfigjson
+                data:
+                  .dockerconfigjson: e30K
+                ---
+                #@overlay/match by=overlay.subset({"kind":"Deployment","metadata":{"name":"source-controller"}}),expects="1+"
+                ---
+                spec:
+                  template:
+                    spec:
+                      #@overlay/match missing_ok=True
+                      imagePullSecrets:
+                      - name: canonical-registry-secret
```

Comment:
I can't find any documentation on what this secret is for and how it's added to the cluster. I understand that secretgen injects the canonical-secret-above with .... for .... is what should be documented.

Reply:
You are right!, there is not use for that secret since all the images the project is going to use are public. That conf come from TAP because some of the images where hosted in private repos!

Let me do a quick fix for that!